src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -522,8 +522,18 @@ internal static string RemoveGuidFromMessage(string message, out bool matchPatte
         internal static string RemoveIdentifierInfoFromMessage(string message, out bool matchPattern)
         {
             matchPattern = false;
-            if (string.IsNullOrEmpty(message))
+            if (message is null ||
+                message.Length < 36 ||
+                message[33] != '[' ||
+                message[32] != ':' ||
+                message[20] != '-' ||
+                message[16] != '-' ||
+                message[12] != '-' ||
+                message[8] != '-')
+            {
+                // Fast return if the message does not start with 'GUID:[]:'.
```

Comment:
```suggestion
                message.Length < 40 ||
                message[8] != '-' ||
                message[13] != '-' ||
                message[18] != '-' ||
                message[23] != '-' ||
                message[36] != ':' ||
                message[37] != '[')
            {
                // Fast return if the message would not match pattern'.
```
Fix several off-by-one errors.